### PR TITLE
Updated Mongo

### LINKED ISSUE #228: Update Mongo
TODO:

* [Update Mongo server version in Singularity](https://github.com/AMP-SCZ/dpdash/blob/a42f0ed1b8e76c7426e25051ad8e25c833b25dc4/singularity/Singularity#L46-L53) to the most recent version
* Update the Mongo version in the package.json to the most recent version

### PR DESCRIPTION
**Pull request by Elkrival**
* Updated Mongo in singularity file
* Updated local version of Mongo

Deployed singularity image and verified that it works 

Screen Shot 2022-11-11 at 9 17 09 PM

### CODE REVIEW

**Review by WMahoney09 — COMMENTED** [review]
👍 This is jumping major versions so there _could be_ breaking changes. Have you verified everything still works on this new version locally?

**Comment by Elkrival** [comment]
> 👍 This is jumping major versions so there _could be_ breaking changes. Have you verified everything still works on this new version locally?

Thanks for this

**Comment by Elkrival** [comment]
> 👍 This is jumping major versions so there _could be_ breaking changes. Have you verified everything still works on this new version locally?

@WMahoney09 https://github.com/AMP-SCZ/dpdash/pull/231/commits/dec48d59fb1c5eeb136a033b01bf65bb7db114c6

File: server/routes/charts.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { Router } from 'express'
-import { ObjectID } from 'mongodb'
+import { ObjectId } from 'mongodb'
```
Comment (tashrifbillah) [inline server/routes/charts.js:2]: Does the new `mongodb` library no longer have `ObjectID` as spelt before? @Elkrival

Comment (Elkrival) [reply server/routes/charts.js:2]: Yes ObjectID is deprecated and now it is ObjectId

**Review by WMahoney09 — APPROVED** [review]
lgtm

File: singularity/configure.py [hunk 2]
```diff
@@ -306,10 +306,10 @@ def configure_supervisord(supervisor_id, config_path):
...
 
 def configure_dpdash(ca_path, key_path, cert_path, dpdash_port, mongo_port, rabbit_port, mongo_password, rabbit_password, session_secret, config_path, data_repo, dpdash_path, rabbit_host, mongo_host, secret):
-    configuration = '''import { readFileSync } from "fs";
-const cert = readFileSync("%(cert_path)s");
-const key = readFileSync("%(key_path)s");
-const ca = [readFileSync("%(ca_path)s")];
+    configuration = '''import path from 'path';
+const cert = path.normalize(("%(cert_path)s");
+const key = path.normalize(("%(key_path)s");
+const ca = path.normalize(("%(ca_path)s");
```
Comment (tashrifbillah) [inline singularity/configure.py:314]: This four lines essentially fixed MGB's connectivity issue.

**Review by tashrifbillah — COMMENTED** [review]
Tashrif's summary of changes:

* Changes in `local-signup.js`, `local-login.js`, and `ldap.js` are beautification related. The semicolons that have been removed--we inserted them through our beautification a while back.

* `charts.js` and `EditChart.react.js` are entirely developed by Gnar company. They will have full control over their changes.

* Changes in `configure.py` helped sort out MGB's connectivity issue.

* `index.js` has changes `ObjectID`-->`ObjectId`

**Review by tashrifbillah — CHANGES_REQUESTED** [review]
However, this error still remains at the backend:

```js
Apptainer> cat ~/.pm2/logs/www-error.log 
MongoServerError: ns not found
    at Connection.onMessage (/sw/apps/dpdash/node_modules/mongodb/lib/cmap/connection.js:207:30)
    at MessageStream.<anonymous> (/sw/apps/dpdash/node_modules/mongodb/lib/cmap/connection.js:60:60)
    at MessageStream.emit (node:events:394:28)
    at processIncomingData (/sw/apps/dpdash/node_modules/mongodb/lib/cmap/message_stream.js:132:20)
    at MessageStream._write (/sw/apps/dpdash/node_modules/mongodb/lib/cmap/message_stream.js:33:9)
    at writeOrBuffer (node:internal/streams/writable:389:12)
    at _write (node:internal/streams/writable:330:10)
    at MessageStream.Writable.write (node:internal/streams/writable:334:10)
    at TLSSocket.ondata (node:internal/streams/readable:747:22)
    at TLSSocket.emit (node:events:394:28)
    at addChunk (node:internal/streams/readable:312:12)
    at readableAddChunk (node:internal/streams/readable:287:9)
    at TLSSocket.Readable.push (node:internal/streams/readable:226:10)
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23) {
  ok: 0,
  code: 26,
  codeName: 'NamespaceNotFound',
  [Symbol(errorLabels)]: Set(0) {}
}

```

Though its effect on overall app function is unclear at this time, can we (attempt to) solve it before merging this PR?

**Comment by Elkrival** [comment]
Is the the entire error log, could this be an issue retained from when the app did not work

**Comment by Elkrival** [comment]
> However, this error still remains at the backend:
> 
> ```js
> Apptainer> cat ~/.pm2/logs/www-error.log 
> MongoServerError: ns not found
>     at Connection.onMessage (/sw/apps/dpdash/node_modules/mongodb/lib/cmap/connection.js:207:30)
>     at MessageStream.<anonymous> (/sw/apps/dpdash/node_modules/mongodb/lib/cmap/connection.js:60:60)
>     at MessageStream.emit (node:events:394:28)
>     at processIncomingData (/sw/apps/dpdash/node_modules/mongodb/lib/cmap/message_stream.js:132:20)
>     at MessageStream._write (/sw/apps/dpdash/node_modules/mongodb/lib/cmap/message_stream.js:33:9)
>     at writeOrBuffer (node:internal/streams/writable:389:12)
>     at _write (node:internal/streams/writable:330:10)
>     at MessageStream.Writable.write (node:internal/streams/writable:334:10)
>     at TLSSocket.ondata (node:internal/streams/readable:747:22)
>     at TLSSocket.emit (node:events:394:28)
>     at addChunk (node:internal/streams/readable:312:12)
>     at readableAddChunk (node:internal/streams/readable:287:9)
>     at TLSSocket.Readable.push (node:internal/streams/readable:226:10)
>     at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23) {
>   ok: 0,
>   code: 26,
>   codeName: 'NamespaceNotFound',
>   [Symbol(errorLabels)]: Set(0) {}
> }
> ```
> 
> Though its effect on overall app function is unclear at this time, can we (attempt to) solve it before merging this PR?

@tashrifbillah Could you help me by identifying what effect this error has on the application? or how to replicate the issue? Is it possible that it could be an error from prior iterations?

File: singularity/Singularity [hunk 3]
```diff
@@ -22,7 +22,7 @@ From:centos:7.4.1708
     yum -y groupinstall "Development tools"
     yum -y install epel-release
     yum -y install net-tools telnet wget git tar which \
-        whereis bzip2 p7zip vim vi vixie-cron cronie sudo
+        bzip2 p7zip vim vi cronie sudo
```
Comment (mikestone14) [inline singularity/Singularity:25]: what's the reason for the changes in this file?

Comment (tashrifbillah) [reply singularity/Singularity:25]: Those packages do not exist anymore!

**Review by mikestone14 — APPROVED** [review]
Based on this comment in Slack from Tashrif, I believe we can now approve & merge this PR.

> I do not have any more logs to provide beyond what I have already provided. Since things are unclear, I shall maintain that the error happens only once when a new database is created. It does not happen during future stop and start of the server. That said, you may merge the PR whenever it is ready.

**Review by tashrifbillah — APPROVED** [review]
